### PR TITLE
Add UMF_BUILD_TESTS option (ON by default)

### PR DESCRIPTION
**Pull request by ldorau**
**Fix required by oneCCL.**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,7 +71,10 @@ install(
     EXPORT ${PROJECT_NAME}-targets)
 
 add_subdirectory(src)
-add_subdirectory(test)
+# do not include tests inside oneCCL
+if(NOT CCL_MAJOR_VERSION)
```
Comment (igchor) [inline CMakeLists.txt:75]: I think it would be better to have a generic "UMF_BUILD_TESTS" option that oneCCL (and anyone else) can switch

Comment (bratpiorka) [reply CMakeLists.txt:75]: agree - please set this option to true by default and I will disable it in oneCCL

Comment (ldorau) [reply CMakeLists.txt]: Done

**Comment by ldorau** [comment]
@bratpiorka @igchor Done

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**